src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_VALUE = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[INITIAL_VALUE];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+        checkSize();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkForIndex(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkForIndex(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkForIndex(index);
+        T arrays = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
+        return arrays;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == element || element != null && element.equals(array[i])) {
+                remove(i);
+                return element;
+            }
+        }
+        throw new NoSuchElementException("No such element exists");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public boolean checkForIndex(int index) {
```

Comment:
Let's make this method **void**. And we should change access modifiers to private if we use them only in the class.